packages/express/lib/utils.js

```diff
@@ -19,17 +20,37 @@ function getStatsFromFile() {
   return stats || {};
 }
 
-function defaultCallback(error) {
+function getPort(host, port) {
```

Comment:
It might make sense to limit the number of ports being tried, i.e. servers being started..:

```javascript
const findPort = (ip, port, max) =>
  new Promise((resolve, reject) => {
    max = max || Math.min(65535, port + 50);
    if (port > max) {
      reject(new Error('unable to find free port'));
    } else {
      const server = createNetServer().unref();
      server.on('error', () => resolve(findPort(ip, port + 1, max)));
      server.listen(port, ip, () => server.close(() => resolve(port)));
    }
  });

const getPort = (ip, port) =>
  findPort(ip, ...(Array.isArray(port) ? port : [port || 8080, port]));
```

https://github.com/untool/untool/blob/master/packages/express/lib/run.js#L24-L37